store/store_action.go

```diff
@@ -564,3 +579,47 @@ func findRev(needle snap.Revision, haystack []snap.Revision) bool {
 	}
 	return false
 }
+
+func reportFetchAssertionsError(res *snapActionResult, assertq AssertionQuery) error {
+	// prefer to report the most unexpected error
+	e := -1
+	errl := res.ErrorList
+	carryingRef := func(i int) bool {
+		aType := asserts.Type(errl[i].Type)
+		return aType != nil && len(errl[i].PrimaryKey) == len(aType.PrimaryKey)
+	}
+	for i, ent := range errl {
```

Comment:
Could you please add a short comment about the error priorities here & what is considered most unexpected? Could the loop break early under certain condition?